### PR TITLE
Feature/datamodel

### PR DESCRIPTION
**Pull request by konpap99**
This PR adds all the required models for the database
The created schemas are the following:
schema

IMPORTANT NOTE: Minor change to index.ts! Server shouldn't start if not connected to database properly

### CODE REVIEW

**Review by nerdsupremacist — APPROVED** [review]

File: src/models/Appointment.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    patientNotes: {
+        type: String,
+    },
+    patientRef: {
+        required: true,
+        type: MiniPatientSchema,
+    },
+    selectedServices: {
+        required: true,
+        type: MiniServiceSchema,
```
[the remark this answers is not in the document]
Comment (Pellumbengineer) [reply src/models/Appointment.ts:97]: @nerdsupremacist  in one specific appointment  do we plan to have more than one service ? if so then yes i also think it should be an array.

**Review by Pellumbengineer — COMMENTED** [review]
It looks good.

File: src/models/Appointment.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (nerdsupremacist) [reply src/models/Appointment.ts:97]: Yeah. I think we should support multiple services per appointment

[the remark this answers is not in the document]
Comment (konpap99) [reply src/models/Appointment.ts:98]: Yep thats true! I missed the brackets

File: src/index.ts [hunk 2]
```diff
@@ -20,12 +22,35 @@ app.use(
...
 app.get('/secret', authenticationRequired, (req, res) => {
     // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
     const id = req.authenticated!.id;
     return res.send(`Secret for user ${id}: 42`);
 });
 
-app.listen(PORT, () => {
-    console.log(`⚡️[server]: Server is running at https://localhost:${PORT}`);
+// database connection
+const dbURI = 'mongodb://dogtorAdmin:DogtorFun!@snf-883170.vm.okeanos.grnet.gr:27017/dogtorDB';
```
[the remark this answers is not in the document]
Comment (konpap99) [reply src/index.ts:61]: Yeah I saw a nightmare about this yesterday :-)

**Pull request merged**